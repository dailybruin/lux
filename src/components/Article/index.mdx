---
name: Article
route: /article
---

import { Playground, PropsTable } from 'docz'
import Article from '.'

# Article

An article's text and multimedia.

Note that `Article` expects an array of `Content`, which

<PropsTable of={Article} />

<Playground>
  <Article dropcap={true} content={[
  {
    type: 'text',
    value:
      'Lorem <a href="https://dailybruin.com">ipsum</a> dolor sit amet, consectetur adipiscing elit. Cras porttitor metus in pellentesque bibendum. Fusce rutrum sollicitudin finibus. Proin pellentesque leo enim, sed ultrices diam laoreet ut. Vivamus ut consequat nulla. Aenean sodales enim ut sapien rhoncus, in rhoncus metus mattis. Phasellus sed turpis tristique, tristique augue quis, lobortis est. Proin consectetur volutpat lacus, vitae luctus orci eleifend ac. Aliquam porttitor aliquet purus. Fusce condimentum eu sapien ut malesuada. Sed at nibh purus. Quisque et cursus nisl. Donec sed tincidunt ex.',
  },
  {
    type: "image",
    url: "https://dailybruin.com/images/2018/10/web.sp_.mwp_.nbk_.ADX_-640x461.jpg",
    caption: "Freshman attacker Ashworth Molthen has registered 21 goals on the season, finding the back of the net in 13 of the 16 games so far this year for UCLA menâ€™s water polo. UCLA is one of two teams remaining in the MPSF conference that remains undefeated.",
    credit: "Amy Dixon/Photo Editor"
  }
]} />
</Playground>
