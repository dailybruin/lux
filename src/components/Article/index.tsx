import * as React from 'react'
import { css } from 'react-emotion'
import Image, { ImageProps } from '../Image'

enum ContentType {
  Text = 'text',
  Image = 'image',
}

interface Text {
  value: string
}

/**
 * Article properties.
 */
interface ArticleProps {
  /** The paragraphs of content for the story. */
  content: any[]
  dropcap: boolean
}

/** A footer to go at the bottom of every page. */
export default class Article extends React.Component<ArticleProps> {
  public static defaultProps = { dropcap: false }

  public render() {
    const renderedContent = this.props.content.map(
      (content: any, i: number) => {
        switch (content.type) {
          case ContentType.Text:
            const text = content as Text
            return (
              <p
                key={i}
                dangerouslySetInnerHTML={{
                  __html: text.value,
                }}
              />
            )
          case ContentType.Image:
            const image = content as ImageProps
            return <Image key={i} {...image} />
          default:
            break
        }
      }
    )

    return (
      <article
        className={
          this.props.dropcap
            ? css`
                p:first-child:first-letter {
                  float: left;
                  font-size: 5em;
                  line-height: 60px;
                  padding-top: 4px;
                  padding-right: 8px;
                  padding-left: 3px;
                }
              `
            : undefined
        }
      >
        {renderedContent}
      </article>
    )
  }
}
