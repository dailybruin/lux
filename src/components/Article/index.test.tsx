import * as React from 'react'
import { cleanup, render } from 'react-testing-library'
import Article from '.'

const content = [
  {
    type: 'text',
    value:
      '<strong>Lorem</strong> <a href="https://dailybruin.com">ipsum</a> dolor sit amet, consectetur adipiscing elit. Cras porttitor metus in pellentesque bibendum. Fusce rutrum sollicitudin finibus. Proin pellentesque leo enim, sed ultrices diam laoreet ut. Vivamus ut consequat nulla. Aenean sodales enim ut sapien rhoncus, in rhoncus metus mattis. Phasellus sed turpis tristique, tristique augue quis, lobortis est. Proin consectetur volutpat lacus, vitae luctus orci eleifend ac. Aliquam porttitor aliquet purus. Fusce condimentum eu sapien ut malesuada. Sed at nibh purus. Quisque et cursus nisl. Donec sed tincidunt ex.',
  },
  {
    type: 'image',
    value: `{
      "caption": "Freshman attacker Ashworth Molthen has registered 21 goals on the season, finding the back of the net in 13 of the 16 games so far this year for UCLA menâ€™s water polo. UCLA is one of two teams remaining in the MPSF conference that remains undefeated.", "credit": "Amy Dixon/Photo Editor",  "url": "https://dailybruin.com/images/2018/10/web.sp_.mwp_.nbk_.ADX_-640x461.jpg", "alt": "Ashworth Molthen playing water polo." }`,
  },
]

describe('Article', () => {
  afterEach(cleanup)

  it('Matches its snapshot', () => {
    const { container } = render(<Article content={content} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
