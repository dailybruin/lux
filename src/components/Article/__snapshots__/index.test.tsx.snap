// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Matches its snapshot 1`] = `
.emotion-1 {
  margin-left: auto;
  margin-right: auto;
  max-width: 620px;
  padding: 1rem;
}

.emotion-0 {
  width: 100%;
  margin-bottom: 0;
}

<article
  class="emotion-1"
>
  <p>
    <strong>
      Lorem
    </strong>
     
    <a
      href="https://dailybruin.com"
    >
      ipsum
    </a>
     dolor sit amet, consectetur adipiscing elit. Cras porttitor metus in pellentesque bibendum. Fusce rutrum sollicitudin finibus. Proin pellentesque leo enim, sed ultrices diam laoreet ut. Vivamus ut consequat nulla. Aenean sodales enim ut sapien rhoncus, in rhoncus metus mattis. Phasellus sed turpis tristique, tristique augue quis, lobortis est. Proin consectetur volutpat lacus, vitae luctus orci eleifend ac. Aliquam porttitor aliquet purus. Fusce condimentum eu sapien ut malesuada. Sed at nibh purus. Quisque et cursus nisl. Donec sed tincidunt ex.
  </p>
  <figure>
    <img
      alt="Ashworth Molthen playing water polo."
      class="emotion-0"
      src="https://dailybruin.com/images/2018/10/web.sp_.mwp_.nbk_.ADX_-640x461.jpg"
    />
    <figcaption>
      Freshman attacker Ashworth Molthen has registered 21 goals on the season, finding the back of the net in 13 of the 16 games so far this year for UCLA menâ€™s water polo. UCLA is one of two teams remaining in the MPSF conference that remains undefeated.
       (
      Amy Dixon/Photo Editor
      )
    </figcaption>
  </figure>
  <h2>
    Custom Header!
  </h2>
</article>
`;
